integration_test/tests/odbc_2_tests.rs

```diff
@@ -0,0 +1,318 @@
+                assert_eq!(
+                    SqlReturn::SUCCESS,
+                    SQLGetData(
+                        stmt as HStmt,
+                        2,
+                        CDataType::SLong,
+                        output_buffer as Pointer,
+                        (BUFFER_LENGTH * std::mem::size_of::<u16>() as i16) as Len,
+                        null_mut()
+                    )
+                );
+                assert_eq!(*(output_buffer as *mut i16), datatype.0);
+            }
+
+            assert_eq!(SqlReturn::NO_DATA, SQLFetch(stmt as HStmt));
+
+            // test that SQLGetTypeInfo works properly with odbc 2.x date type
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetTypeInfo(stmt as HStmt, SqlDataType::EXT_TIMESTAMP)
+            );
+            assert_eq!(SqlReturn::SUCCESS, SQLFetch(stmt as HStmt));
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetData(
+                    stmt as HStmt,
+                    2,
+                    CDataType::SLong,
+                    output_buffer as Pointer,
+                    (BUFFER_LENGTH * std::mem::size_of::<u16>() as i16) as Len,
+                    null_mut()
+                )
+            );
+            assert_eq!(*(output_buffer as *mut i16), 11);
+
+            // test that SQLGetTypeInfo returns an error for odbc 3.x date type
+            assert_eq!(
+                SqlReturn::ERROR,
+                SQLGetTypeInfo(stmt as HStmt, SqlDataType::TIMESTAMP)
+            );
+        }
+    }
+
+    #[test]
+    fn test_data_retrieval() {
+        let env_handle = setup();
+        connect(env_handle);
+        let env_handle: HEnv = setup();
+        let (conn_handle, _, _, _) = connect(env_handle);
+        let mut stmt: Handle = null_mut();
+
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLAllocHandle(
+                    HandleType::Stmt,
+                    conn_handle as *mut _,
+                    &mut stmt as *mut Handle
+                )
+            );
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetTypeInfo(stmt as HStmt, SqlDataType::UNKNOWN_TYPE)
+            );
+
+            // query for date, which is stored as a string. We'll then convert to date using the odbc 2 type
+            let mut query: Vec<WideChar> = cstr::to_widechar_vec("select `stardate` from class");
+            query.push(0);
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLExecDirectW(stmt as HStmt, query.as_ptr(), NTS as SmallInt as i32),
+                "{}",
+                get_sql_diagnostics(HandleType::Stmt, stmt as Handle)
+            );
+
+            // fetch date field as odbc 2.x TimeStamp
+            fetch_and_get_data(
+                stmt,
+                Some(5),
+                vec![SqlReturn::SUCCESS; 1],
+                vec![CDataType::TimeStamp],
```

Comment:
The C type didn't change between version right?
I think what you want to check is that you are getting the correct SQL type for the column in the resultset by calling SQLColAttribute with SQL_DESC_TYPE

Reply:
yes, you are right about sqlcolumns, and that's partially why this pr went up, so SQLColumns can be added as part of 1688. I can update the comment to be a little clearer

Reply:
What I mean is that this test is kind of useless because there was not change affecting getData as far as I can tell. We go straight from bson data to C data and we skip any kind of SQL mapping.

Reply:
I see what you mean now, that makes sense about the bson -> c conversion, and you are right that we didn't update the c types. I can remove it then

Reply:
Cool, thank you!